inlong-sort/sort-connectors/src/main/java/org/apache/inlong/sort/flink/hive/formats/orc/RowVectorizer.java

```diff
@@ -150,9 +158,141 @@ private static void setColumn(int rowId, ColumnVector column, LogicalType type,
                 vector.set(rowId, timestamp);
                 break;
             }
-            // TODO: support MAP type
+            case ARRAY: {
+                Object[] inputArray = (Object[]) field;
+                ColumnVector innerVector = constructColumnVectorFromArray(inputArray);
+                ListColumnVector tempListVector = new ListColumnVector(1, innerVector);
+                tempListVector.lengths[0] = inputArray.length;
+                ListColumnVector vector = (ListColumnVector) column;
+                vector.setElement(rowId, 0, tempListVector);
+                break;
+            }
+            case MAP: {
+                Map<?, ?> mapField = (Map<?, ?>) field;
+                int mapEleSize = mapField.size();
+                Object[] keys = new Object[mapEleSize];
+                Object[] values = new Object[mapEleSize];
+                int i = 0;
+                for (Map.Entry<?, ?> entry : mapField.entrySet()) {
+                    keys[i] = entry.getKey();
+                    values[i] = entry.getValue();
+                    ++i;
+                }
+
+                ColumnVector keysVector = constructColumnVectorFromArray(keys);
+                ColumnVector valuesVector = constructColumnVectorFromArray(values);
+                MapColumnVector tempMapVector = new MapColumnVector(1, keysVector, valuesVector);
+                tempMapVector.lengths[0] = mapEleSize;
+
+                MapColumnVector vector = (MapColumnVector) column;
+                vector.setElement(rowId, 0, tempMapVector);
+                break;
+            }
             default:
                 throw new UnsupportedOperationException("Unsupported type: " + type);
         }
     }
+
+    @VisibleForTesting
+    static ColumnVector constructColumnVectorFromArray(Object[] inputArray) {
```

Comment:
Do not get ColumnVector from like this. Could you infer child type from schema provided from manager?